pylon/src/nginx.conf.template

```diff
@@ -33,20 +33,35 @@ http {
   #       can be simply set to 127.0.0.1.
   resolver 127.0.0.1;
 
+  # Turn gzip on
+  map $request_uri $routing_accept_encoding {
+    default "";
+    ~*/static "gzip";
+  }
+
   server {
     listen      80;
     server_name localhost;
+    client_max_body_size 256M;
+    client_body_buffer_size 256M;
+    proxy_connect_timeout 60m;
+    proxy_send_timeout 60m;
+    proxy_read_timeout 60m;
+    send_timeout 60m;
 
+    #
     # General-purpose reverse proxy.
-    location ~ ^/r/(http|https)/([^/]+)/(\d*)(.*)$ {
-      set $target_scheme $1;
-      set $target_host   $2;
-      set $target_port   :$3;
-      set $target_path   $4$is_args$args;
+    # The proxy to WebHDFS API depends on it.
+    #
+
+    location ~ ^/a/([^/]+)/(\d*)(.*)$ {
```

Comment:
>/a [](start = 16, length = 2)

Why change from "/r" to "/a"?

Reply:
I think `/a` looks more like a temporary functionality that will eventually be retired.

Reply:
Sure, we should retire this general proxy or hide it for internal usage.

---
In reply to: [175000956](https://github.com/Microsoft/pai/pull/377#discussion_r175000956) [](ancestors = 175000956)